packages/transformers/elm/src/ElmTransformer.js

```diff
@@ -0,0 +1,128 @@
+// @flow
+import {Transformer} from '@parcel/plugin';
+import commandExists from 'command-exists';
+import path from 'path';
+import fs from 'fs';
+import spawn from 'cross-spawn';
+import elm from 'node-elm-compiler';
+import {inject as injectElmHMR} from 'elm-hot';
+import {minify} from 'terser';
+import ThrowableDiagnostic from '@parcel/diagnostic';
+
+export default new Transformer({
+  async transform({asset, options}) {
+    const config = {
+      cwd: path.dirname(asset.filePath),
+      debug: !(
+        options.env.PARCEL_ELM_NO_DEBUG || options.mode === 'production'
+      ),
+      optimize: asset.env.minify,
+    };
+
+    (await elm.findAllDependencies(asset.filePath)).forEach(filePath =>
+      asset.addIncludedFile({filePath}),
+    );
+
+    let code = await compileToString(asset, options, config);
+    if (options.hot) code = injectElmHMR(code);
+    if (config.optimize) code = minifyElmOutput(code);
+
+    asset.type = '.js';
+    asset.setCode(code);
+    return [asset];
+  },
+});
+
+async function compileToString(asset, options, config) {
+  const installPackage = name => {
+    if (!options.autoinstall) {
+      throw new ThrowableDiagnostic({
+        diagnostic: {
+          message: `Denpendency '${name}' is not installed and autoinstall is turned off. Either install dependency manually or enable autoinstall`,
+          origin: '@parcel/package-manager',
+        },
+      });
+    }
+
+    return options.packageManager.resolve(name, asset.filePath);
+  };
+  const pathToElmBin = await pathToElm(options.projectRoot, installPackage);
+  await ensureElmJson(asset, pathToElmBin);
+
+  const compileOptions = {
+    cwd: config.cwd,
+    debug: config.debug,
+    optimize: config.optimize,
+    pathToElm: pathToElmBin,
+  };
+  return elm.compileToString(asset.filePath, compileOptions);
+}
+
+async function pathToElm(root, installPackage) {
+  if (!commandExists.sync('elm')) {
+    const elmBin = path.resolve(root, 'node_modules/elm/bin/elm');
+
+    if (!fs.existsSync(elmBin)) await installPackage('elm');
+    return elmBin;
+  }
+
+  return undefined; // use globally installed elm
+}
```

Comment:
This would break with Yarn PnP, it should be something like this:
```js
async function ensureElmInstalled(asset, options) {
  if (!commandExists.sync('elm')) {
    await options.packageManager.resolve('elm/package.json', asset.filePath, {
      autoinstall: options.autoinstall,
    });
  }
}
```
and then you spawn `elm` while relying on npm/npx/Yarn to have injected the bin folder into the PATH.


I'm also still getting errors from `@parcel/package-manager` (`@parcel/transformer-elm: Cannot find module 'elm' from '.../elm/src'`), there is probably still some code where the `/package.json` part of `elm/package.json` is removed incorrectly.

Reply:
@mischnic could autoinstall be optional for elm until someone implements it? It's a bit a shame this whole PR is stalled for that feature which the author doesn't care much about (I think that's understandable).

Reply:
Yes, let's do that